app/src/lib/local-storage.ts

```diff
@@ -87,3 +87,44 @@ export function getNumber(
 export function setNumber(key: string, value: number) {
   localStorage.setItem(key, value.toString())
 }
+
+/**
+ * Retrieve an array of `number` values from a given local
+ * storage entry, if found. The array will be empty if the
+ * key doesn't exist or if the values cannot be converted
+ * into numbers
+ *
+ * @param key local storage entry to read
+ */
+export function getNumberArray(key: string): ReadonlyArray<number> {
+  const numbersAsText = localStorage.getItem(key)
+  let values = new Array<number>()
+  if (numbersAsText !== null) {
+    try {
+      values = numbersAsText
+        .split(NumberArrayDelimiter)
+        .map(n => parseInt(n, 10))
+        .filter(n => !isNaN(n))
+    } catch {
+      // clear array, just in case
+      values.splice(-1)
```

Comment:
I'm not wild about this. There's no way the above block can throw and assign something to the values array. Since we're only using the value to return and since we're not doing any post processing I think this code can be made more clear by not allocating this mutable value. Something like

```ts
return (localStorage.getItem(key) || '')
  .split(NumberArrayDelimiter)
  .map(parseFloat)
  .filter(n => !isNaN(n))
```

None of these methods can throw so we can get rid of the try catch and there's always going to be an empty array to return if the values couldn't be parsed.